fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -714,14 +714,17 @@ class Record implements Type {
          */
         private final boolean isNullable;
 
+        @Nullable
+        private final String name;
```

Comment:
Sure thing, last thing we want is to pollute API design, on the other hand, the type repository's internal representation logic is very valuable and well tested, it would be a shame to have to rewrite it because of that. Let's have a discussion about that.